notebooks/responsibleaidashboard/text/responsibleaidashboard-DBPedia-text-classification-model-debugging.ipynb

```diff
@@ -0,0 +1,361 @@
+   "source": [
+    "Fetch a pre-trained huggingface model on the DBPedia dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "36e20709",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DBPEDIA_MODEL_NAME = \"dbpedia_model\"\n",
+    "NUM_LABELS = 9\n",
+    "\n",
+    "class FetchModel(object):\n",
+    "    def __init__(self):\n",
+    "        pass\n",
+    "\n",
+    "    def fetch(self):\n",
+    "        zipfilename = DBPEDIA_MODEL_NAME + '.zip'\n",
+    "        url = ('https://publictestdatasets.blob.core.windows.net/models/' +\n",
+    "               DBPEDIA_MODEL_NAME + '.zip')\n",
+    "        urlretrieve(url, zipfilename)\n",
+    "        with zipfile.ZipFile(zipfilename, 'r') as unzip:\n",
+    "            unzip.extractall(DBPEDIA_MODEL_NAME)\n",
+    "\n",
+    "def retrieve_dbpedia_model():\n",
+    "    fetcher = FetchModel()\n",
+    "    action_name = \"Model download\"\n",
+    "    err_msg = \"Failed to download model\"\n",
+    "    max_retries = 4\n",
+    "    retry_delay = 60\n",
+    "    retry_function(fetcher.fetch, action_name, err_msg,\n",
+    "                   max_retries=max_retries,\n",
+    "                   retry_delay=retry_delay)\n",
+    "    model = AutoModelForSequenceClassification.from_pretrained(\n",
+    "        DBPEDIA_MODEL_NAME, num_labels=NUM_LABELS)\n",
+    "    return model\n",
+    "\n",
+    "model = retrieve_dbpedia_model()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ea0eef05",
+   "metadata": {},
+   "source": [
+    "Load the model and tokenizer"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c93f3d7e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tokenizer = AutoTokenizer.from_pretrained(\"bert-base-uncased\")\n",
+    "\n",
+    "device = -1\n",
+    "if device >= 0:\n",
+    "    model = model.cuda()\n",
+    "\n",
+    "# build a pipeline object to do predictions\n",
+    "pred = pipeline(\n",
+    "    \"text-classification\",\n",
+    "    model=model,\n",
+    "    tokenizer=tokenizer,\n",
+    "    device=device,\n",
+    "    return_all_scores=True\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9d54a252",
+   "metadata": {},
+   "source": [
+    "Define the encoded classes, which needs to come from the original trained model, and display the number of errors on the dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2178e47f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ml_wrappers import wrap_model\n",
+    "wrapped_model = wrap_model(pred, test_data, 'text_classification')\n",
+    "\n",
+    "encoded_classes = ['Agent', 'Device', 'Event', 'Place', 'Species',\n",
+    "                   'SportsSeason', 'TopicalConcept', 'UnitOfWork',\n",
+    "                   'Work']\n",
+    "labels = [encoded_classes.index(y) for y in test_data['label'].tolist()]\n",
```

Comment:
```suggestion
    "labels = test_data['label'].apply(lambda y: encoded_classes[y]).tolist()\n",
```
Faster than list comprehension due to numpy/C optimizations with pandas .apply under the hood afaik.

Reply:
I think the linter doesn't allow us to use lambdas